scripts/github-ci.sh

```diff
@@ -22,24 +22,18 @@ if [ "$OS_NAME" == "linux" ]; then
     # CI (https://github.com/actions/checkout/issues/760)
     docker run --privileged \
            -v $HOME/.gradle:/root/.gradle \
-           -v ${GITHUB_BUILD_DIR}:/opt/go/${GO_SRC_DIR}/ \
+           -v ${GITHUB_BUILD_DIR}:/root/go/${GO_SRC_DIR}/ \
            -i "${CI_IMAGE}" \
-           bash -c "git config --global --add safe.directory \$GOPATH/${GO_SRC_DIR} && make -C \$GOPATH/${GO_SRC_DIR} ${WHAT}"
+           bash -c "git config --global --add safe.directory \$(go env GOPATH)/${GO_SRC_DIR} && make -C \$(go env GOPATH)/${GO_SRC_DIR} ${WHAT}"
 fi
 
 # The following is executed only on macOS machines.
 if [ "$OS_NAME" == "osx" ]; then
     # GitHub CI installs Go and Qt directly in the macos action, before executing
     # this script.
     go version
-    export GOPATH=~/go
-    export PATH="$PATH:~/go/bin"
-    mkdir -p $GOPATH/$(dirname $GO_SRC_DIR)
-    # GitHub checkout action (git clone) seem to require current work dir
-    # to be the root of the repo during its clean up phase. So, we push it
-    # here and pop in the end.
-    pushd ../ && cp -a bitbox-wallet-app $GOPATH/$(dirname $GO_SRC_DIR)
-    cd $GOPATH/$GO_SRC_DIR
-    make "$WHAT"
-    popd
+    export PATH="~/go/bin:$PATH"
+    mkdir -p $(go env GOPATH)/$(dirname $GO_SRC_DIR)
+    cp -a ../bitbox-wallet-app $(go env GOPATH)/$(dirname $GO_SRC_DIR)
```

Comment:
this seems wrong, there should be no need to copy the repo somewhere else.

Reply:
Is that something from how go used to work? that the go code had to be copied to a certain directory structure?

Reply:
There is this comment earlier:

> # Because we need to compile some Go code without modules,
> # the source must be placed in a specific directory as expected by Go.
> # The path is relative to GOPATH.
> GO_SRC_DIR=src/github.com/BitBoxSwiss/bitbox-wallet-app